testing/test-nx-utils/src/lib/utils/tree.integration.test.ts

```diff
@@ -0,0 +1,32 @@
+import { join } from 'node:path';
+import { createTreeWithEmptyWorkspace } from 'nx/src/generators/testing-utils/create-tree-with-empty-workspace';
+import { describe, expect, it } from 'vitest';
+import { readJsonFile } from '@code-pushup/utils';
+import { materializeTree } from './tree';
+
+describe('materializeTree', () => {
+  const baseDir = join('tmp', 'materialize-tree');
+
+  it('should create files from tree', async () => {
+    const root = join(baseDir, 'materialize');
+
+    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
```

Comment:
🧪 Output cleanup: While the `tmp` folder is ignored by default, each test should clean up after itself (the strategy is on our [Wiki](https://github.com/code-pushup/cli/wiki/Testing-Strategy#test-outputs)). I recommend adding an `after` hook that removes the `materialize` subfolder to ensure clean state.

Reply:
Good catch, I missed that :see_no_evil: Yes, the test should clean up the file system changes after it runs :+1: